### PR TITLE
Add a new project root-signing-staging

### LINKED ISSUE #345: New project root-signing-staging
As documented in https://github.com/sigstore/root-signing/issues/929 I'd like to start maintaining the staging TUF repository for root-signing with the TUF-on-CI tool. The reasons are 
* have a meaningful way to test architectural and process changes in root signing  
* If root-signing-staging works well, start maintaining the actual root-signing repository with TUF-on-CI 

Currently the "staging TUF repo" is just static files within root-signing git -- a setup that doesn't really resemble production very much. The plan is to actually maintain the staging repository act more like the production one (with some changes like lower signature thresholds to keep maintenance manageable).

I plan to open some PRs to this repo and link to this issue (as I figure out Pulumi etc), but high level plan is this:
* define a sigstore-rootsigning-team which could include at least @jku (me) and @kommendorkapten
* define a new project root-signing-staging (I will take other name suggestions though)
* Configure the GCP KMS like production root-signing, so that the new project can sign with a testing key (preferably a new one for this specific purpose)
* (likely later) Configure GCS for publishing like production root-signing 

cc @haydentherapper

### PR DESCRIPTION
**Pull request by jku**
#### Summary

Add new repository root-signing-staging, see #345 for details.

Fixes #345 (although further tweaks may be required: we'll see how the bot permissions match what tuf-on-ci expects, and what is needed to configure Pages publishing from GH actions).

### CODE REVIEW

**Review by bobcallaway — APPROVED** [review]

File: github-sync/github-data/sigstore/repositories.yaml [hunk 1]
```diff
@@ -1410,6 +1410,67 @@ repositories:
...
           - sigstore-keyholders
+  - name: root-signing-staging
+    owner: sigstore
+    description: "Staging TUF repository for Sigstore trust root"
+    homepageUrl: ""
+    defaultBranch: main
+    allowAutoMerge: false
+    allowMergeCommit: true
+    allowRebaseMerge: false
+    allowSquashMerge: false
```
Comment (haydentherapper) [inline github-sync/github-data/sigstore/repositories.yaml:1421]: Do you want to allow squash merges? I typically disable merge commits to keep the history clean, allow squash, and allow auto merge.

File: github-sync/github-data/sigstore/repositories.yaml [hunk 2]
```diff
@@ -1410,6 +1410,67 @@ repositories:
...
+    archived: false
+    autoInit: false
+    deleteBranchOnMerge: false
```
Comment (haydentherapper) [inline github-sync/github-data/sigstore/repositories.yaml:1424]: Does automation need a static branch, or does it create new branches frequently? Might want this on.

File: github-sync/github-data/sigstore/repositories.yaml (hunk 1 above)
Comment (jku) [reply github-sync/github-data/sigstore/repositories.yaml:1421]: Not 100% sure but I don't think I do...commits in signing event PRs are meaningful as they come from multiple sources(different signers, repository workflow, etc). Merging them would look confusing IMO

File: github-sync/github-data/sigstore/repositories.yaml (hunk 2 above)
Comment (jku) [reply github-sync/github-data/sigstore/repositories.yaml:1424]: Not that frequently, but a branch per signing event. I would maybe rather start with not deleting branches for easier forensics

**Comment by jku** [comment]
Gentle ping on this one. Like I said I'm not 100% confident this configuration is exactly correct but it's as close as I can get it: I believe we need to try and see how it goes. If you have questions, let me know.

cc @cpanato who seems to do a lot around here or I guess the  tsc folks @bobcallaway @trevrosen @lukehinds @priyawadhwa @SantiagoTorres ?

**Comment by trevrosen** [comment]
Yeah this looks fine, let's give it a shot and iterate as we need to.

**Review by trevrosen — APPROVED** [review]

**Comment by haydentherapper** [comment]
Can a TSC member merge this?

**Pull request merged**